MoreLinq/Lead.cs

```diff
@@ -37,9 +38,13 @@ public static partial class MoreEnumerable
         /// <param name="resultSelector">A projection function which accepts the current and subsequent (lead) element (in that order) and produces a result</param>
         /// <returns>A sequence produced by projecting each element of the sequence with its lead pairing</returns>
 
-        public static IEnumerable<TResult> Lead<TSource, TResult>(this IEnumerable<TSource> source, int offset, Func<TSource, TSource, TResult> resultSelector)
+        public static IEnumerable<TResult> Lead<TSource, TResult>(this IEnumerable<TSource> source, int offset, Func<TSource, TSource?, TResult> resultSelector)
         {
-            return Lead(source, offset, default!, resultSelector);
+            if (source is null) throw new ArgumentNullException(nameof(source));
```

Comment:
Consistency: `is null` vs `== null` in L66-68. Probably prefer `is null`, but I'm good either way.

Reply:
You're right, it's inconsistent, but since it probably exists in many places, it's best to roll it into a separate PR.